src/services/consent/services/consentVersionService.js

```diff
@@ -0,0 +1,98 @@
+const { authenticate } = require('@feathersjs/authentication');
+const { BadRequest } = require('@feathersjs/errors');
+const {
+	iff,
+	isProvider,
+	disallow,
+} = require('feathers-hooks-common');
+
+const {
+	populateCurrentSchool,
+	restrictToCurrentSchool,
+	hasPermission,
+} = require('../../../hooks');
+
+const { modelServices: { prepareInternalParams } } = require('../../../utils');
+
+// todo check userId // <-- TODO: was bedeutet der todo *-*
+const ConsentVersionServiceHooks = {
+	before: {
+		all: [iff(isProvider('external'), [
+			authenticate('jwt'),
+			populateCurrentSchool, // TODO: test if it is needed
+			restrictToCurrentSchool, // TODO restricted erscheint mir hier nicht hilfreich
+		])],
+		find: [],
+		get: [],
+		create: [iff(isProvider('external'), hasPermission('SCHOOL_EDIT'))],
+		update: [disallow()],
+		patch: [disallow()],
+		remove: [disallow()],
+	},
+	after: {
+		all: [],
+		find: [],
+		get: [],
+		create: [],
+		update: [],
+		patch: [],
+		remove: [],
+	},
+};
+
+class ConsentVersionService {
+	constructor(options) {
+		this.options = options || {};
+		this.docs = {};
+	}
+
+	createBase64File(data = {}) {
+		const { schoolId, consentData } = data;
+		if (consentData) {
+			if (!schoolId) {
+				return Promise.reject(new BadRequest('SchoolId is required for school consents.'));
+			}
+			return this.app.service('base64Files').create({ schoolId, data: consentData, fileType: 'pdf' });
+		}
+		return Promise.resolve({});
+	}
+
+	get(id, params) {
+		return this.app.service('consentVersionsModel').get(id, prepareInternalParams(params));
+	}
+
+	find(params) {
+		const { query } = params;
+		if (query.schoolId) {
+			if (!query.$or) {
+				query.$or = [];
+			}
+			query.$or.push({ schoolId: query.schoolId });
+			query.$or.push({ schoolId: undefined }); // TODO test if undefined work
```

Comment:
is this done?

Reply:
I've removed the conditions to add values without schoolId. We need Policies for the current school only.

Reply:
No it do not work for old consents.

Reply:
If schoolId is undefined than the query is where schoolId is undefined and it returns all values without schoolId.